cupy/core/_routines_math.pyx

```diff
@@ -519,9 +519,10 @@ cpdef scan_core(ndarray a, axis, scan_op op, dtype=None, ndarray out=None):
         result[...] = a
 
     if axis is None:
-        result = result.ravel()
         for accelerator in _accelerator._routine_accelerators:
             if accelerator == _accelerator.ACCELERATOR_CUB:
+                if result is None:
+                    result = a.astype(dtype=dtype).ravel()
```

Comment:
```suggestion
                    result = a.astype(dtype=dtype, copy=False).ravel()
```
Can we avoid copy when a.dtype == dtype or is the copy needed here? (I have not studied this code closely)

Reply:
Thanks!  I had assumed that `copy=False` was the default behavior... 😓